File: server/php/src/Application/Domain/Payments/PaymentDetails/OperationPaymentReferences.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class OperationPaymentReferences
{
    public ?string $merchantReference = null;

    public function toArray(): array
    {
        return [
            'merchantReference' => $this->merchantReference
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/operation_payment_references.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class OperationPaymentReferences
          attr_accessor :merchant_reference

          def initialize(merchant_reference: nil)
            @merchant_reference = merchant_reference
          end

          def to_h
            {
              'merchantReference' => @merchant_reference
            }
          end
        end
      end
    end
  end
end